polyply/src/generate_templates.py

```diff
@@ -92,40 +92,25 @@ def find_interaction_involving(block, current_node, prev_node):
                'linking atom {} and atom {}.')
         raise IOError(msg.format(block.nodes[0]["resname"], prev_node, current_node))
 
-def _expand_inital_coords(block):
+def _expand_inital_coords(block, bond=None, pos=None, fixed=None,
+                          iterations=50, weight="weight", max_box=1.0):
     """
-    Given a `block` generate initial random coordinates
-    for all atoms in the block. Note that the initial
-    coordinates though random, have a defined distance
-    correspodning to a bond , constaint or virtual-site.
+    Given a `graph` generate initial random coordinates
+    in three dimensions and relax them using a Fruchterman
+    Reingold generic simulation to relax the coordinates
+    topologically.
 
     Parameters
     -----------
-    block:   :class:`vermouth.molecule.Block`
+    block:   nx.Graph
 
     Returns
     ---------
     dict
       dictonary of node index and position
     """
-    coords = {}
-    #TODO this should actually be the index
-    atom = list(block.nodes)[0]
-    coords[atom] = np.array([0, 0, 0])
-
-    vectors = norm_sphere(values=1000)
-    for prev_node, current_node in nx.dfs_edges(block, source=atom):
-        prev_coord = coords[prev_node]
-        is_vs, interaction, inter_type = find_interaction_involving(block,
-                                                                    current_node,
-                                                                    prev_node)
-        if is_vs:
-            coords[current_node] = construct_vs(inter_type, interaction, coords)
-        else:
-            coords[current_node], _ = _take_step(vectors,
-                                                 float(interaction.parameters[1]),
-                                                 prev_coord)
-    return coords
+    return nx.spring_layout(block, dim=3, k=bond, pos=pos, fixed=fixed,
+                            iterations=iterations, weight=weight, scale=max_box)
```

Comment:
I've also had decent results with `planar_layout` (not applicable to all molecules) and `kamada_kawai_layout`. You could consider using those as alternatives, or use them as initial coordinates (`pos` argument) for `spring_layout`

Reply:
This was actually for 2D. For 3D spring_layout may be far superior.

Reply:
You mean the other two methods are more suited for 2D? Since Kamada_Kwai in fact performs an optimization of a target function I think it is overkill for initial positions.